sql/2020/14_PWA/sw_events.sql

```diff
@@ -0,0 +1,25 @@
+#standardSQL
+# SW events - based on 2019/14_03.sql
+SELECT
+  client,
+  event,
+  COUNT(DISTINCT page) AS freq,
+  total,
+  COUNT(DISTINCT page) / total AS pct
+FROM
+  `httparchive.almanac.service_workers`
+JOIN
+  (SELECT client, COUNT(DISTINCT page) AS total FROM `httparchive.almanac.service_workers` WHERE date = '2020-08-01' GROUP BY client)
+USING (client),
+  UNNEST(ARRAY_CONCAT(
+    REGEXP_EXTRACT_ALL(body, r'\.on(install|activate|fetch|push|notificationclick|notificationclose|sync|canmakepayment|paymentrequest|message|messageerror)\s*='),
```

Comment:
Really need custom metrics for these next year 😞
No change needed now.

Reply:
Luckily this only comes from the `service_workers` table so small.